tests/bot_tests/gameplay_framework.py

```diff
@@ -96,27 +104,23 @@ def step(self):
         # get/set orders
         for bot in self.bots:
             if hasattr(bot, "orders"):
-                orders = bot.orders
+                orders = ret_dict["orders"]
                 if orders is not None:
-                    self.game.set_orders(power_name=bot.power_name, orders=orders.get_list_of_orders())
+                    self.game.set_orders(power_name=bot.power_name, orders=orders)
 
         self.cur_local_message_round+=1
 
         self.game.process()
         return {"messages": msgs_to_send}, self.game.is_game_done
             
-            
-
-            
-        
 
+@gen.coroutine            
+def game_loop():
+    game_play_obj = GamePlay(None, [RandomProposerBot_AsyncBot, RandomProposerBot_AsyncBot, RandomProposerBot_AsyncBot, SmartOrderAccepterBot, SmartOrderAccepterBot, SmartOrderAccepterBot, SmartOrderAccepterBot], 3, True)
```

Comment:
this specific test should not be in this file

Reply:
I think this helps those students that have to test-run SOA for the game loop with their new function. They don't need to go to SOA_test.py just yet.